keda/templates/01-serviceaccount.yaml

```diff
@@ -4,11 +4,15 @@ kind: ServiceAccount
 metadata:
   labels:
     app.kubernetes.io/name: {{ .Values.serviceAccount.name }}
+    azure.workload.identity/use: "{{ .Values.podIdentity.azureWorkload.enabled }}"
     {{- include "keda.labels" . | indent 4 }}
-  {{- if .Values.serviceAccount.annotations }}
+  {{- if or .Values.podIdentity.azureWorkload.enabled .Values.serviceAccount.annotations }}
   annotations:
-  {{- toYaml .Values.serviceAccount.annotations | nindent 4}}
-  {{- end }}
+    azure.workload.identity/client-id: {{ .Values.podIdentity.azureWorkload.clientId }}
+    azure.workload.identity/tenant-id: {{ .Values.podIdentity.azureWorkload.tenantId }}
+    azure.workload.identity/service-account-token-expiration: {{ .Values.podIdentity.azureWorkload.tokenExpiration }}
```

Comment:
I would add an if around this to only do it when azure workload was enabled, otherwise this can be confusing.